tests/framework/nodeplacement.go

```diff
@@ -22,18 +22,29 @@ var (
 // The values chosen are valid, but the pod will likely not be schedulable.
 func (f *Framework) TestNodePlacementValues() sdkapi.NodePlacement {
 	nodes, _ := f.K8sClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
-	affinityTestValue = &v1.Affinity{
-		NodeAffinity: &v1.NodeAffinity{
-			RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
-				NodeSelectorTerms: []v1.NodeSelectorTerm{
-					{
-						MatchExpressions: []v1.NodeSelectorRequirement{
-							{Key: "kubernetes.io/hostname", Operator: v1.NodeSelectorOpIn, Values: []string{nodes.Items[0].Name}},
+
+	nodeIdx := -1
+	for i, node := range nodes.Items {
+		if _, hasLabel := node.Labels["node-role.kubernetes.io/worker"]; hasLabel {
+			nodeIdx = i
+			break
+		}
+	}
+
+	if nodeIdx > -1 {
```

Comment:
Are we ok with using empty `affinityTestValue` when nodeIdx is -1?

Reply:
fixed, just for consistency.